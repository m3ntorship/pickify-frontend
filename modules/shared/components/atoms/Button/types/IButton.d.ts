import type * as EButton from './EButton';

declare namespace IButton {
  export interface IProps extends IHandlers {
    [property: string]: unknown;
    buttonText?: string | null;
    disabled?: boolean;
    variant: EButton.buttonVariantValues;
    size: EButton.buttonSizeValues;
    leftIcon?: boolean;
    rightIcon?: boolean;
    onlyIcon?: boolean;
    buttonType?: 'button' | 'reset' | 'submit';
  }

  export interface IHandlers {
    onClick?: () => void;
  }
}
export { IButton };
