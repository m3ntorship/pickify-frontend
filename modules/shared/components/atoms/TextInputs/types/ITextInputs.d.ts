import type * as ETextInput from './ETextInput';

declare namespace ITextInputs {
  export interface IProps extends IData {
    [property: string]: unknown;
    label?: string;
    variants: ETextInput.Variants;
    inputType: ETextInput.InputType;
    disabled?: boolean;
    letter?: string;
  }

  export interface IData {
    id: string;
  }
}
export { ITextInputs };
