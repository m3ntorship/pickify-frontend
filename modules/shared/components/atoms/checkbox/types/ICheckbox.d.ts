declare namespace ICheckbox {
  export interface ICheckBoxProps {
    disabled: boolean;
    size: 'Default' | 'Small';
    onMockClick?: () => void;
  }
  export interface ICheckBoxSvgProps {
    style: string;
  }
}
export { ICheckbox };
