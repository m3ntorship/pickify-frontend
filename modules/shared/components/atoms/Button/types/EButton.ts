export enum buttonVariantValues {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  QUATERNARY = 'quaternary',
  TEXT = 'text',
  LIKE = 'like',
  DISLIKE = 'dislike',
  DELETE = 'delete',
}
export enum buttonSizeValues {
  FULLWIDTH = 'fullWidth',
  XLARGE = 'xlarge',
  LARGE = 'large',
  MEDIUM = 'medium',
  NORMAL = 'normal',
  SMALL = 'small',
}
