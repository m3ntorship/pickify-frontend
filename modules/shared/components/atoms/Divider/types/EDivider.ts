export enum DividerType {
  Horizontal = 'horizontal',
  Vertical = 'vertical',
}
