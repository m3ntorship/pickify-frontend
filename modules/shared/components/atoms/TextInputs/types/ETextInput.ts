export enum Variants {
  Default = 'default',
  Error = 'error',
  Success = 'success',
}
export enum InputType {
  Choices = 'choices',
  Default = 'default',
  LeftIcon = 'left-icon',
  PrefixDropdown = 'prefix-dropdown',
  Prefix = 'prefix',
  RightIcon = 'right-icon',
}
