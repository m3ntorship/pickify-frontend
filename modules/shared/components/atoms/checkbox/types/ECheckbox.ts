export enum Size {
  Default = 'Default',
  Small = 'Small',
}
