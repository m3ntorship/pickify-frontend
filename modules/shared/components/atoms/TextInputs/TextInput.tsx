import React from 'react';
import className from 'classnames';
import type { FC, ReactElement } from 'react';
import DeleteIcon from '../../icons/inputFocus.svg';
import ErrorIcon from '../../icons/inputError.svg';
import SuccessIcon from '../../icons/inputSuccess.svg';
import ArrowDownIcon from '../../icons/inputArrowDown.svg';
import HomeIcon from '../../icons/inputHome.svg';
import styles from './TextInput.module.css';
import type { ITextInputs } from './types/ITextInputs';
import * as ETextInput from './types/ETextInput';

const TextInput: FC<ITextInputs.IProps> = ({
  label,
  variants,
  inputType,
  disabled,
  letter,
  ...props
}): ReactElement => {
  const [inputVal, setInputVal] = React.useState<string>('');

  const changeHandler = (e: React.FormEvent<HTMLInputElement>): void => {
    setInputVal(e.currentTarget.value);
  };

  const hideIconHandler = (): void => {
    setInputVal('');
  };

  const inputClasses: string = className(
    styles['form-input'],
    {
      [styles.error]: variants === ETextInput.Variants.Error && !disabled,
      [styles.success]: variants === ETextInput.Variants.Success && !disabled,
    },
    {
      [styles['input-with-right-icon']]:
        inputType === ETextInput.InputType.RightIcon,
      [styles['input-with-left-icon']]:
        inputType === ETextInput.InputType.LeftIcon,
      [styles['input-with-prefix']]: inputType === ETextInput.InputType.Prefix,
      [styles['input-with-dropdown-prefix']]:
        inputType === ETextInput.InputType.PrefixDropdown,
      [styles['input-with-choices']]:
        inputType === ETextInput.InputType.Choices,
    },
  );

  /*
    when it comes to authentication we need to validate
  */
  // const inputId = id.trim().split(' ').join('');

  return (
    <div className={styles['form-group']}>
      {label ? (
        <label htmlFor={props.id} className={styles['form-label']}>
          {label}
        </label>
      ) : (
        ' '
      )}

      <div className={styles['form-control']}>
        <input
          className={inputClasses}
          data-testid="text-input"
          type="text"
          id={props.id}
          disabled={disabled}
          onChange={changeHandler}
          value={inputVal}
          data-variant={variants}
          data-input-type={inputType}
          {...(props as unknown)}
        />
        <span className={styles['status-icon']}>
          {variants === ETextInput.Variants.Default &&
            !disabled &&
            inputVal && (
              <DeleteIcon onClick={hideIconHandler} data-testid="delete-icon" />
            )}
          {variants === ETextInput.Variants.Error && !disabled && <ErrorIcon />}
          {variants === ETextInput.Variants.Success && !disabled && (
            <SuccessIcon />
          )}
        </span>
        <span className={styles['control-icon']}>
          {inputType === ETextInput.InputType.RightIcon && <ArrowDownIcon />}
          {inputType === ETextInput.InputType.LeftIcon && <HomeIcon />}
          {inputType === ETextInput.InputType.Prefix && '+20'}
          {inputType === ETextInput.InputType.PrefixDropdown && (
            <select className={styles.prefixes} disabled={disabled}>
              <option value="+20">+20</option>
              <option value="+50">+50</option>
            </select>
          )}
          {inputType === ETextInput.InputType.Choices && (
            <span className="letter">{letter}</span>
          )}
        </span>
      </div>
    </div>
  );
};

export default TextInput;
