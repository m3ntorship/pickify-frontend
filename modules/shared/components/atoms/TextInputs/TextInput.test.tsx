import React from 'react';
import { render, screen } from '@testing-library/react';
import * as renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import type { TargetElement } from '@testing-library/user-event';
import TextInput from './TextInput';
import * as ETextInput from './types/ETextInput';

describe('TextInput', () => {
  it('should render TextInput with DeleteIcon when we pass filled value', () => {
    const onChange = jest.fn();
    const onClick = jest.fn();

    render(
      <TextInput
        label="label"
        id="my label"
        inputType={ETextInput.InputType.Default}
        variants={ETextInput.Variants.Default}
        disabled={false}
        value="aaa"
        placeholder="Type something..."
        onChangeInputValueHandler={onChange}
        onClickDeleteInputValueHandler={onClick}
        onBlurInputHandler={(): boolean => true}
      />,
    );

    const deleteIcon: TargetElement = screen.getByTestId('delete-icon');

    expect(deleteIcon).toBeInTheDocument();
  });

  it('should call onChange Handler', () => {
    const onChange = jest.fn();
    const onClick = jest.fn();
    const calledTwice = 2;

    render(
      <TextInput
        label="label"
        id="my label"
        inputType={ETextInput.InputType.Default}
        variants={ETextInput.Variants.Default}
        disabled={false}
        value=""
        onChangeInputValueHandler={onChange}
        onClickDeleteInputValueHandler={onClick}
        onBlurInputHandler={(): boolean => true}
        placeholder="Type something..."
      />,
    );

    const input: TargetElement = screen.getByTestId('text-input');

    userEvent.type(input, 'aa');

    expect(onChange).toHaveBeenCalledTimes(calledTwice);
  });

  it('should call onClick Handler', () => {
    const onChange = jest.fn();
    const onClick = jest.fn();
    const calledOnce = 1;

    render(
      <TextInput
        label="label"
        id="my label"
        inputType={ETextInput.InputType.Default}
        variants={ETextInput.Variants.Default}
        disabled={false}
        value="aaa"
        placeholder="Type something..."
        onChangeInputValueHandler={onChange}
        onClickDeleteInputValueHandler={onClick}
        onBlurInputHandler={(): boolean => true}
      />,
    );

    const input: TargetElement = screen.getByTestId('delete-icon');

    userEvent.click(input);

    expect(onClick).toHaveBeenCalledTimes(calledOnce);
  });

  it('should call onBlur Handler', () => {
    const onChange = jest.fn();
    const onClick = jest.fn();
    const onBlur = jest.fn();
    const calledOnce = 1;

    render(
      <TextInput
        label="label"
        id="my label"
        inputType={ETextInput.InputType.Default}
        variants={ETextInput.Variants.Default}
        disabled={false}
        value=""
        placeholder="Type something..."
        onChangeInputValueHandler={onChange}
        onClickDeleteInputValueHandler={onClick}
        onBlurInputHandler={onBlur}
      />,
    );

    const input: TargetElement = screen.getByTestId('text-input');

    userEvent.click(input);
    userEvent.tab();

    expect(onBlur).toHaveBeenCalledTimes(calledOnce);
  });

  it('should render TextInput with variant [error]', () => {
    const funHandler = jest.fn();

    const tree = renderer.create(
      <TextInput
        label="label"
        id="my label"
        inputType={ETextInput.InputType.Default}
        variants={ETextInput.Variants.Error}
        disabled={false}
        value=""
        placeholder="Type something..."
        onChangeInputValueHandler={funHandler}
        onClickDeleteInputValueHandler={funHandler}
        onBlurInputHandler={(): boolean => true}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render TextInput with variant and input type [default]', () => {
    const funHandler = jest.fn();

    const tree = renderer.create(
      <TextInput
        label="label"
        id="my label"
        inputType={ETextInput.InputType.Default}
        variants={ETextInput.Variants.Default}
        disabled={false}
        value=""
        placeholder="Type something..."
        onChangeInputValueHandler={funHandler}
        onClickDeleteInputValueHandler={funHandler}
        onBlurInputHandler={(): boolean => true}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render TextInput with variant [success]', () => {
    const funHandler = jest.fn();

    const tree = renderer.create(
      <TextInput
        label="label"
        id="my label"
        inputType={ETextInput.InputType.Default}
        variants={ETextInput.Variants.Success}
        disabled={false}
        value=""
        placeholder="Type something..."
        onChangeInputValueHandler={funHandler}
        onClickDeleteInputValueHandler={funHandler}
        onBlurInputHandler={(): boolean => true}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render TextInput with input type [left-icon]', () => {
    const funHandler = jest.fn();

    const tree = renderer.create(
      <TextInput
        label="label"
        id="my label"
        inputType={ETextInput.InputType.LeftIcon}
        variants={ETextInput.Variants.Default}
        disabled={false}
        value=""
        placeholder="Type something..."
        onChangeInputValueHandler={funHandler}
        onClickDeleteInputValueHandler={funHandler}
        onBlurInputHandler={(): boolean => true}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render TextInput with input type [prefix-dropdown]', () => {
    const funHandler = jest.fn();

    const tree = renderer.create(
      <TextInput
        label="label"
        id="my label"
        inputType={ETextInput.InputType.PrefixDropdown}
        variants={ETextInput.Variants.Default}
        disabled={false}
        value=""
        placeholder="Type something..."
        onChangeInputValueHandler={funHandler}
        onClickDeleteInputValueHandler={funHandler}
        onBlurInputHandler={(): boolean => true}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render TextInput with input type [prefix]', () => {
    const funHandler = jest.fn();

    const tree = renderer.create(
      <TextInput
        label="label"
        id="my label"
        inputType={ETextInput.InputType.Prefix}
        variants={ETextInput.Variants.Default}
        disabled={false}
        value=""
        placeholder="Type something..."
        onChangeInputValueHandler={funHandler}
        onClickDeleteInputValueHandler={funHandler}
        onBlurInputHandler={(): boolean => true}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render TextInput with input type [right-icon]', () => {
    const funHandler = jest.fn();

    const tree = renderer.create(
      <TextInput
        label="label"
        id="my label"
        inputType={ETextInput.InputType.RightIcon}
        variants={ETextInput.Variants.Default}
        disabled={false}
        value=""
        placeholder="Type something..."
        onChangeInputValueHandler={funHandler}
        onClickDeleteInputValueHandler={funHandler}
        onBlurInputHandler={(): boolean => true}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render TextInput with input type [choices]', () => {
    const funHandler = jest.fn();

    const tree = renderer.create(
      <TextInput
        label="label"
        id="my label"
        inputType={ETextInput.InputType.Choices}
        variants={ETextInput.Variants.Default}
        disabled={false}
        value=""
        placeholder="Type something..."
        onChangeInputValueHandler={funHandler}
        onClickDeleteInputValueHandler={funHandler}
        onBlurInputHandler={(): boolean => true}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
