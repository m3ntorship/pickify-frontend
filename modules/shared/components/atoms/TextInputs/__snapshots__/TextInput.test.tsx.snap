// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render TextInput with input type [choices] 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="my label"
  >
    label
  </label>
  <div
    className="form-control"
  >
    <input
      className="form-input input-with-choices"
      data-input-type="choices"
      data-testid="text-input"
      data-variant="default"
      dir="auto"
      disabled={false}
      id="my label"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Type something..."
      type="text"
      value=""
    />
    <span
      className="status-icon"
    >
      
    </span>
    <span
      className="control-icon"
    >
      <span
        className="letter"
      />
    </span>
  </div>
</div>
`;

exports[`TextInput should render TextInput with input type [left-icon] 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="my label"
  >
    label
  </label>
  <div
    className="form-control"
  >
    <input
      className="form-input input-with-left-icon"
      data-input-type="left-icon"
      data-testid="text-input"
      data-variant="default"
      dir="auto"
      disabled={false}
      id="my label"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Type something..."
      type="text"
      value=""
    />
    <span
      className="status-icon"
    >
      
    </span>
    <span
      className="control-icon"
    >
      <svg
        data-file-name="SvgInputHome"
      />
    </span>
  </div>
</div>
`;

exports[`TextInput should render TextInput with input type [prefix] 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="my label"
  >
    label
  </label>
  <div
    className="form-control"
  >
    <input
      className="form-input input-with-prefix"
      data-input-type="prefix"
      data-testid="text-input"
      data-variant="default"
      dir="auto"
      disabled={false}
      id="my label"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Type something..."
      type="text"
      value=""
    />
    <span
      className="status-icon"
    >
      
    </span>
    <span
      className="control-icon"
    >
      +20
    </span>
  </div>
</div>
`;

exports[`TextInput should render TextInput with input type [prefix-dropdown] 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="my label"
  >
    label
  </label>
  <div
    className="form-control"
  >
    <input
      className="form-input input-with-dropdown-prefix"
      data-input-type="prefix-dropdown"
      data-testid="text-input"
      data-variant="default"
      dir="auto"
      disabled={false}
      id="my label"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Type something..."
      type="text"
      value=""
    />
    <span
      className="status-icon"
    >
      
    </span>
    <span
      className="control-icon"
    >
      <select
        className="prefixes"
        disabled={false}
      >
        <option
          value="+20"
        >
          +20
        </option>
        <option
          value="+50"
        >
          +50
        </option>
      </select>
    </span>
  </div>
</div>
`;

exports[`TextInput should render TextInput with input type [right-icon] 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="my label"
  >
    label
  </label>
  <div
    className="form-control"
  >
    <input
      className="form-input input-with-right-icon"
      data-input-type="right-icon"
      data-testid="text-input"
      data-variant="default"
      dir="auto"
      disabled={false}
      id="my label"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Type something..."
      type="text"
      value=""
    />
    <span
      className="status-icon"
    >
      
    </span>
    <span
      className="control-icon"
    >
      <svg
        data-file-name="SvgInputArrowDown"
      />
    </span>
  </div>
</div>
`;

exports[`TextInput should render TextInput with variant [error] 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="my label"
  >
    label
  </label>
  <div
    className="form-control"
  >
    <input
      className="form-input error"
      data-input-type="default"
      data-testid="text-input"
      data-variant="error"
      dir="auto"
      disabled={false}
      id="my label"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Type something..."
      type="text"
      value=""
    />
    <span
      className="status-icon"
    >
      <svg
        data-file-name="SvgInputError"
      />
    </span>
    <span
      className="control-icon"
    />
  </div>
</div>
`;

exports[`TextInput should render TextInput with variant [success] 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="my label"
  >
    label
  </label>
  <div
    className="form-control"
  >
    <input
      className="form-input success"
      data-input-type="default"
      data-testid="text-input"
      data-variant="success"
      dir="auto"
      disabled={false}
      id="my label"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Type something..."
      type="text"
      value=""
    />
    <span
      className="status-icon"
    >
      <svg
        data-file-name="SvgInputSuccess"
      />
    </span>
    <span
      className="control-icon"
    />
  </div>
</div>
`;

exports[`TextInput should render TextInput with variant and input type [default] 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="my label"
  >
    label
  </label>
  <div
    className="form-control"
  >
    <input
      className="form-input"
      data-input-type="default"
      data-testid="text-input"
      data-variant="default"
      dir="auto"
      disabled={false}
      id="my label"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Type something..."
      type="text"
      value=""
    />
    <span
      className="status-icon"
    >
      
    </span>
    <span
      className="control-icon"
    />
  </div>
</div>
`;
